app/scripts/modules/core/src/pipeline/config/triggers/pipelineRoles.component.ts

```diff
@@ -0,0 +1,46 @@
+import { IController, module } from 'angular';
+import { AuthenticationService } from 'core/authentication';
+
+class PipelineRolesController implements IController {
+  public pipeline: any;
+  public allowedRoles: string[];
+
+  public $onInit(): void {
+    this.allowedRoles = AuthenticationService.getAuthenticatedUser().roles;
+    this.pipeline.roles = this.pipeline.roles || [];
+  }
+
+  public toggleCheck(role: string) {
+    if (this.pipeline.roles.indexOf(role) === -1) {
```

Comment:
It's a little more idiomatic to use `if (this.pipeline.roles.includes(role))`, though the boolean is flipped in that case.